### PR TITLE
Fixes #1545 - Make it easy to run integration tests

### LINKED ISSUE #1545: Make it easy to run integration/scale tests via Docker
At Mesosphere we run the integration tests via docker on our CI system. We would like to allow everyone with a docker installation to run these tests easily on their development machine.

What is more, we reuse the regular Dockerfile for integration testing which results in duplicate/non-optimal builds. Thus we would like to provide a tailored Dockerfile basis for integration testing and a simple way to run integration tests/scaling tests with this file.

### PR DESCRIPTION
**Pull request by kolloch**
via a script and docker.

Adjust test settings:

* to be sensitive to whether sbt formatting is enabled
* to include durations
* to print stack traces in error summary

Make scaling test slightly more agressive.

### CODE REVIEW

**Comment by kolloch** [comment]
I already enabled downloading the test scripts from this PR to TeamCity.

So this patch kind off builds itself.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 .idea
 *.iml
+docker-volumes
```
Comment (aquamatthias) [inline .gitignore]: Why not put this under target?

File: Dockerfile.build-base [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM mesosphere/mesos:0.22.1-1.0.ubuntu1404
```
Comment (aquamatthias) [inline Dockerfile.build-base]: build-base sounds like, it will be reused/refined somewhere?

File: bin/run-tests.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+# be verbose and print all commands, better for debugging
+set -x
+
+PROJECT_DIR=`dirname $0`/..
+PROJECT_DIR=$(cd "$PROJECT_DIR"; pwd)
+
+if [ -z "$1" ]; then
+    echo "Usage:" >&2
+    echo "$0 <unique build id without crazy characters or spaces>" >&2
```
Comment (aquamatthias) [inline bin/run-tests.sh]: Would it make sense to create a buildId, if it is omitted?
As I started it the first time, my question was: where do I get the buildId....

File: bin/run-tests.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+if ! docker build -t marathon-buildbase:$BUILD_ID -f "$PROJECT_DIR/Dockerfile.build-base" "$PROJECT_DIR"; then
+    fatal "Build for the buildbase failed" >&2
+fi
+
+docker run \
+    --rm=true \
+    --name marathon-itests-$BUILD_ID \
+    -v "$BUILD_VOLUME_DIR/sbt:/root/.sbt" \
+    -v "$BUILD_VOLUME_DIR/ivy2:/root/.ivy2" \
+    -v "$BUILD_VOLUME_DIR/targets/main:/marathon/target" \
```
Comment (aquamatthias) [inline bin/run-tests.sh]: For local tests: can we change that in a way, that only changes need to get recompiled?

File: bin/run-tests.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+if ! docker build -t marathon-buildbase:$BUILD_ID -f "$PROJECT_DIR/Dockerfile.build-base" "$PROJECT_DIR"; then
+    fatal "Build for the buildbase failed" >&2
+fi
+
+docker run \
+    --rm=true \
+    --name marathon-itests-$BUILD_ID \
+    -v "$BUILD_VOLUME_DIR/sbt:/root/.sbt" \
+    -v "$BUILD_VOLUME_DIR/ivy2:/root/.ivy2" \
```
Comment (aquamatthias) [inline bin/run-tests.sh]: Can we make it configurable, so that I can use ~/.ivy2 and ~/.sbt

File: project/build.scala [hunk 6]
```diff
@@ -47,12 +47,32 @@ object MarathonBuild extends Build {
...
+   * U - unformatted mode
+   * I - show reminder of failed and canceled tests without stack traces
+   * T - show reminder of failed and canceled tests with short stack traces
+   * G - show reminder of failed and canceled tests with full stack traces
+   * K - exclude TestCanceled events from reminder
+   *
+   * http://scalatest.org/user_guide/using_the_runner
+   */
+  lazy val formattingTestArg = Tests.Argument(if (withFormatting) "-eDST" else "-eWUDST")
+
```
Comment (aquamatthias) [inline project/build.scala]: Great comment!

File: bin/run-tests.sh [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+cleanup
+trap cleanup EXIT
+
+if ! docker build -t marathon-buildbase:$BUILD_ID -f "$PROJECT_DIR/Dockerfile.build-base" "$PROJECT_DIR"; then
+    fatal "Build for the buildbase failed" >&2
+fi
+
+docker run \
+    --rm=true \
```
Comment (bobrik) [inline bin/run-tests.sh]: `--rm` works without args

File: .gitignore (hunk 1 above)
Comment (kolloch) [inline .gitignore]: Because I wanted it not to be gone after an sbt clean.

We talked about it and I leave this the default but allow configuration.

File: Dockerfile.build-base (hunk 2 above)
Comment (kolloch) [inline Dockerfile.build-base]: Currently, no. It would potentially make sense to use this for building the assembly of the official image as well but as far as I know two phase builds are not supported on dockerhub,

File: bin/run-tests.sh (hunk 3 above)
Comment (kolloch) [inline bin/run-tests.sh]: I tried to document it and we just use "test" as default.

File: bin/run-tests.sh (hunk 4 above)
Comment (kolloch) [inline bin/run-tests.sh]: I made everything configurable in my local version.

File: bin/run-tests.sh (hunk 5 above)
Comment (kolloch) [inline bin/run-tests.sh]: I made everything configurable in my local version.

File: project/build.scala (hunk 6 above)
Comment (kolloch) [inline project/build.scala]: Thanks!

**Comment by kolloch** [comment]
Made stuff configurable.

**Comment by aquamatthias** [comment]
This is cool!
